### PR TITLE
Upgrade K8s dependencies to 1.17.6.

### PR DESCRIPTION
**Pull request by Harwayne**
## Proposed Changes

- Upgrade K8s dependencies to 1.17.6.
  - This corresponds to https://github.com/knative/eventing/pull/3163.

**Release Note**

```release-note
The minimum supported Kubernetes version is now 0.16.
```

### CODE REVIEW

**Comment by nachocano** [comment]
I saw a new script to update k8s deps inside hack in eventing.
Do we want that here as well?
https://github.com/knative/eventing/pull/3163/files#diff-cb335268b7e5dc60157f02ea957b88a0

**Comment by Harwayne** [comment]
> I saw a new script to update k8s deps inside hack in eventing.
> Do we want that here as well?
> https://github.com/knative/eventing/pull/3163/files#diff-cb335268b7e5dc60157f02ea957b88a0

I actually used that one, but didn't know if I should pull it over. I decided not to, simply so I wouldn't have to think about the copyright header (do I leave it as knative? change it to Google?).

File: go.mod [hunk 1]
```diff
@@ -80,3 +80,5 @@ replace gopkg.in/yaml.v2 => gopkg.in/yaml.v2 v2.2.2
 replace honnef.co/go/tools => honnef.co/go/tools v0.0.1-2019.2.3
 
 replace sigs.k8s.io/yaml => sigs.k8s.io/yaml v1.1.0
+
+replace k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.17.6
```
Comment (grantr) [inline go.mod:84]: Move this to the larger replace block since it's not likely to be temporary.

Comment (Harwayne) [reply go.mod]: Done.

**Comment by Harwayne** [comment]
/retest

This seems to be failing with `error processing import paths in "/home/prow/go/src/github.com/google/knative-gcp/test/test_images/auditlogs_receiver/pod.yaml": error resolving image references: exit status 1` and I don't know why. That file isn't changed.

**Pull request merged**